cypress/e2e/tests/pages/explorer/cluster-dashboard.spec.ts

```diff
@@ -0,0 +1,155 @@
+    clusterDashboard.customBadge().selectCheckbox('Customize cluster icon').set();
+    clusterDashboard.customBadge().clusterIcon().children().should('not.have.class', 'cluster-local-logo');
+    clusterDashboard.customBadge().clusterIcon().contains(settings.iconText.original);
+    clusterDashboard.customBadge().iconText().set(settings.iconText.new);
+    clusterDashboard.customBadge().clusterIcon().contains(settings.iconText.new);
+
+    // Apply Changes
+    clusterDashboard.customBadge().applyAndWait('/v3/clusters/local');
+
+    // check header and side nav for update
+    header.clusterIcon().children().should('have.class', 'cluster-badge-logo');
+    header.clusterName().should('contain', 'local');
+    header.customBadge().should('contain', settings.description.new);
+    const burgerMenu = new BurgerMenuPo();
+
+    burgerMenu.clusters().first().find('span').should('contain', settings.iconText.new);
+
+    // Reset
+    clusterDashboard.addCustomBadge('Edit Cluster Badge').click();
+    clusterDashboard.customBadge().selectCheckbox('Customize cluster icon').set();
+    clusterDashboard.customBadge().selectCheckbox('Show badge for this cluster').set();
+
+    // Apply Changes
+    clusterDashboard.customBadge().applyAndWait('/v3/clusters/local');
+
+    // check header and side nav for update
+    header.clusterIcon().children().should('have.class', 'cluster-local-logo');
+    header.clusterName().should('contain', 'local');
+    header.customBadge().should('not.exist');
+    burgerMenu.clusters().first().find('svg').should('have.class', 'cluster-local-logo');
+  });
+
+  it('can view deployments', () => {
+    clusterDashboard.goTo();
+    cy.getRancherResource('v1', 'apps.deployments', '?exclude=metadata.managedFields').then((resp: Cypress.Response<any>) => {
+      const count = resp.body['count'];
+
+      simpleBox.simpleBox().eq(2).should('contain.text', count).and('contain.text', 'Deployments');
+    }).then((el: any) => {
+      el.click();
+
+      const workloadDeployments = new WorkloadsDeploymentsListPagePo('local', 'apps.deployment');
+
+      workloadDeployments.waitForPage();
+    });
+  });
+
+  it('can view nodes', () => {
+    clusterDashboard.goTo();
+    cy.getRancherResource('v1', 'nodes', '?exclude=metadata.managedFields').then((resp: Cypress.Response<any>) => {
+      const count = resp.body['count'];
+      let text = '';
+
+      if (count > 1) {
+        text = 'Nodes';
+      } else {
+        text = 'Node';
+      }
+      simpleBox.simpleBox().eq(1).should('contain.text', count).and('contain.text', text);
+    }).then((el: any) => {
+      el.click();
+
+      const nodesPage = new NodesPagePo('local');
+
+      nodesPage.waitForPage();
+    });
+  });
+
+  it('can view events', () => {
+    const podName = `e2e-test-${ +new Date() }`;
+
+    // Create a pod to trigger events
+    cy.createPod('local', podName, 'nginx:latest');
```

Comment:
Can this be removed in an afterAll (but only if this line is called... to avoid cases where this test isn't run... pod doesn't exist... delete request fails)

Reply:
I dont think there's a way to run an afterAll step conditionally. Instead, I went ahead and removed the pod within the test itself

Reply:
If the test fails will the removal still run?

If not was thinking something like
```
let removePod = false;
it('can view events', () => {
  // Create a pod to trigger events
  cy.createPod('local', podName, 'nginx:latest').then(() => removePod = true);
...

afterAll() {
  if (removePod) {
     cy.deleteRancherResource('v1', 'pods/local', `pod-${ podName }`);
  }
 ```
 
 if though the delete executes in all cases this isn't needed

Reply:
ah, you have a point. if the test fails the removal will not run. okay, I'll plug this ^ in. Thank you!